pgjdbc/src/main/java/org/postgresql/gss/GSSOutputStream.java

```diff
@@ -73,10 +73,12 @@ public void write(byte[] b, int off, int len) throws IOException {
     }
     // Write out the rest, chunk the writes, so we do not exceed the maximum encrypted message size
     while (len >= buf.length) {
-      writeWrapped(b, off, buf.length);
-      off += buf.length;
-      len -= buf.length;
+      int writeSize = Math.min(buf.length - count, len);
```

Comment:
Why is this needed? I think it could be reverted as well.

Reply:
ok, you are right.

`len >= buf.length` will ensure `min(buf.length - count, len) == buf.length`. The reaming data will be written in the next call.

I also fix a bug when I write the test code. Inside `writeWrapped` use the input length, not use the `b.length`, `b` may comes from a bigger buffer.